lib/chef/resource.rb

```diff
@@ -416,6 +420,10 @@ def only_if(command = nil, opts = {}, &block)
     # @param block [Proc] A ruby block to run. Ignored if a command is given.
     #
     def not_if(command = nil, opts = {}, &block)
+      if command.is_a?(Hash) && opts.empty?
+        opts = command
+        command = nil
+      end
```

Comment:
I agree 100%.  I use ruby 1.9 on my daily, and really just never realized double-splat worked a little differently than the way k:v pairs implicitly load into opts={} .

Thanks for the feedback.